auditwheel/wheel_abi.py

```diff
@@ -30,18 +31,25 @@ def get_wheel_elfdata(wheel_fn: str):
     with InGenericPkgCtx(wheel_fn) as ctx:
         for fn, elf in elf_file_filter(ctx.iter_files()):
             is_py_ext, py_ver = elf_is_python_extension(fn, elf)
-            if is_py_ext:
-                log.info('processing: %s', fn)
-                elftree = lddtree(fn)
-                full_elftree[fn] = elftree
-                for key, value in elf_find_versioned_symbols(elf):
-                    versioned_symbols[key].add(value)
-
-                if py_ver == 2:
-                    uses_ucs2_symbols |= any(
-                        True for _ in elf_find_ucs2_symbols(elf))
-                full_external_refs[fn] = lddtree_external_references(elftree,
-                                                                     ctx.path)
+
+            # Check for invalid binary wheel format: no shared library should be found in purelib
+            so_path_split = fn.split(os.sep)
+            if 'purelib' in so_path_split:
+                raise RuntimeError(('Invalid binary wheel, found shared library "%s" in purelib folder.\n'
+                                    'The wheel has to be platlib compliant in order to be repaired by auditwheel.') %
+                                   so_path_split[-1])
```

Comment:
Issue #32 should be addressed in a separate PR. Let's focus this PR on handling pre-bundled shared libraries properly.